contrib/segmentio/kafka-go/kafka.go

```diff
@@ -0,0 +1,150 @@
+package kafka
+
+import (
+	"context"
+	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+
+	"github.com/segmentio/kafka-go"
+)
+
+// NewReader calls kafka.NewReader and wraps the resulting Consumer.
+func NewReader(conf kafka.ReaderConfig, opts ...Option) *Reader {
+	return WrapReader(kafka.NewReader(conf), opts...)
+}
+
+// NewWriter calls kafka.NewWriter and wraps the resulting Producer.
+func NewWriter(conf kafka.WriterConfig, opts ...Option) *Writer {
+	return WrapWriter(kafka.NewWriter(conf), opts...)
+}
+
+// WrapReader wraps a kafka.Reader so that any consumed events are traced.
+func WrapReader(c *kafka.Reader, opts ...Option) *Reader {
+	wrapped := &Reader{
+		Reader: c,
+		cfg:    newConfig(opts...),
+	}
+	log.Debug("contrib/confluentinc/confluent-kafka-go/kafka: Wrapping Reader: %#v", wrapped.cfg)
+	return wrapped
+}
+
+// A Reader wraps a kafka.Reader.
+type Reader struct {
+	*kafka.Reader
+	cfg  *config
+	prev ddtrace.Span
+}
+
+func (r *Reader) startSpan(ctx context.Context, msg *kafka.Message) ddtrace.Span {
```

Comment:
There seems to be a ton of duplication between here and our other Kafka integration. In order to keep the experience consistent I'd prefer it if we would factor out the duplication into another package, let's say `contrib/internal/kafkautil`. WDYT?

Reply:
I also thought the same, didn't get time to do that.

Reply:
Unfortunately we'll have to do that, it's too much duplication and I don't want to risk discrepancies and unilateral bugs.

Reply:
Ok, will do it when I get time